src/app/joint-control/pose/pose.component.ts

```diff
@@ -0,0 +1,86 @@
+import {Component, OnInit, TemplateRef, ViewChild} from "@angular/core";
+import {FormControl} from "@angular/forms";
+import {NgbModal} from "@ng-bootstrap/ng-bootstrap";
+import {Observable, from, map} from "rxjs";
+import {PoseService} from "src/app/shared/services/pose.service";
+import {Pose} from "src/app/shared/types/pose";
+
+@Component({
+    selector: "app-pose",
+    templateUrl: "./pose.component.html",
+    styleUrls: ["./pose.component.css"],
+})
+export class PoseComponent implements OnInit {
+    @ViewChild("modalContent") modalContent: TemplateRef<any> | undefined;
+
+    poses: Pose[] = [];
+
+    modalTitle = "";
+
+    nameFormControl: FormControl<string> = new FormControl();
+
+    selectedPoseId?: string;
+
+    constructor(
+        private poseService: PoseService,
+        private modalService: NgbModal,
+    ) {}
+
+    ngOnInit(): void {
+        this.poseService
+            .getPosesObservable()
+            .subscribe((poses) => (this.poses = poses));
```

Comment:
Is it necessary to subscribe to the observable and assign the value to a global variable instead of storing a reference to the observable and use the `async` pipe for displaying it in the component?
Supposedly the `async` pipe deals with unsubscribing upon completion/failure automatically, preventing memory leaks.